### DISCUSSION TITLE
Importing from a Python script

### DISCUSSION THREAD

**Discussion by samueltober** [PrefectHQ/prefect#4806] [Q&A] [answered]
Hi! I am trying to run a flow in which i import from a Python file, I am not sure how I should handle this since Prefect cannot find the module upon running. Can I somehow specify this in the Docker image? I am running Prefect cloud with GCP storage.

**Comment by kvnkho** [comment, marked as answer]
Hi @samueltober ,

GCP Storage only keeps track of the script that contains the Flow. If you are importing functions from other scripts, these will not be included during Flow registration. The recommendation is to package your scripts together as a Python module and add them to a Docker container, and then you can use Docker Storage that will have access to these modules as the container will be pulled.

I have a Dockerfile example [here](https://github.com/kvnkho/demos/tree/main/prefect/docker_with_local_storage) that might help with that.

**Reply by samueltober** [reply]
Thank you for your response! I am new to Docker so I do not fully understand your example Dockfile. Am I correct in that it is the setup.py together with pip install -e ., that allows us to import from a separate python file?

**Reply by kvnkho** [reply]
Of course! `setup.py` is that makes the Python scripts into a module. After that is setup you can:

1. Copy over the files to the Docker image with the Dockerfile
2. `pip install -e .` to install it as a module

This would treat it as a module in your container, and then you can import the other files as Python handles the path stuff through the pip install.

I also invite you to join the Slack channel at prefect.io/slack . We regularly answer questions like this there, and you can get faster answers for your future questions. Staying here is perfectly fine though.

**Reply by samueltober** [reply]
Thank you! I will join the slack channel :)
